app/controllers/subscription_controller.rb

```diff
@@ -119,19 +119,20 @@ def digest
   end
 
   def multiple_add
-    unless params[:names]
+    unless params[:tagnames]
       flash[:notice] = "Please enter tags for subscription in the url."
       redirect_to "/subscriptions" + "?_=" + Time.now.to_i.to_s
+      return
     end
-    tag_list = params[:names].split(',')
+    tag_list = params[:tagnames]
```

Comment:
Why are you not splitting the tagnames array?
Is your branch able to subscribe tags when tagnames = `kites,oil,,kites`.
I worry it won't.

Reply:
Because params[:tagnames] is already a comma separated list.

Reply:
Can I have a screenshot of `localhost:3000/subscribe/multiple/tag/kites,oil,,kite` ?
I worry that you will be subscribed to `kites,oil,,kites` instead of `kites`, `oil` and `kite`.

Reply:
I was thinking, we could check for special characters like comma or !, because only these will create problems, otherwise the list is already comma separated, and I think even if we split kites,oil,,kite it would use one comma for splitting but the other one would still be a part of tag_list

Reply:
Sorry I can't understand what you are saying.
Scenario of symbol in tag_name will not happen as tag_names will consists of tags and in the database we have regex for it. So you can ignore it.

Reply:
I was saying that if we encounter any extra comma or something of the same sort, we can check it with an if condition and ignore that iteration. But since this situation won't come like you are saying, then this is good to go , right?

Reply:
If else condition is not require as the code which I implemented is clever enough to handle multiple commas if they appear together. Check out the functional minutest which I wrote.

Reply:
Yes absolutely, so this is done right?

Reply:
yes, I did it right. Please change it to previous state

Reply:
I haven't changed anything here except removing .split(",") , like I said it's already a comma separated list.

Reply:
I think the travis build is not failing because of the .split(',') thing. Because I changed the test from 'blog,kites,,balloon,awesome' to 'blog,kites,balloon,awesome' and it is giving the same error locally.

Reply:
Hi, just chiming in here to try to be helpful. I think we will have to revert back to the version with `.split(',')` because of these lines below:

```ruby
        tag_list.each do |t|
          if t.length.positive?
            tag = Tag.find_by(name: t)
```

unfortunately if we try to do `.each` on a string, it may throw an error like `NoMethodError: undefined method `each' for "one,two":String`. Does that make sense?

Reply:
```suggestion
    tag_list = params[:tagnames].split(',')
```

Reply:
if params[:names].is_a? String
      tag_list = params[:names].split(',') 
    else
      tag_list = params[:names]
    end

Try this